library/std/src/sys/unix/ext/process.rs

```diff
@@ -319,3 +320,70 @@ impl IntoRawFd for process::ChildStderr {
 pub fn parent_id() -> u32 {
     crate::sys::os::getppid()
 }
+
+/// Converts a signal number to a string representation suitable for a human reader.
+///
+/// This does not necessarily produce the same output as the C `strsignal` function.
+/// Currently there are two deviations from the return value of `strsignal`:
+///
+///  * `signal_describe` does not translate the string according to the locale; C `strsignal`
+///    usually does.  Unfortunately a translated version is not currently available.
+///
+///  * `signal_describe` will include the signal abbrevation (eg, `SIGINT`) if
+///    the platform C library can provide that.
+///
+/// The precise form of the output should not be relied on.
+///
+/// # Examples
+///
+/// ```
+/// #![feature(unix_signal_strings)]
+/// use std::os::unix::process::signal_describe;
+/// assert!(signal_describe(9).contains("Killed"));
+/// ```
+#[unstable(feature = "unix_signal_strings", issue = "none")]
+pub fn signal_describe(sig: i32) -> String {
+    let mut buf = String::new();
+    os::signal_display(&mut buf, sig).unwrap();
+    buf
+}
+
+/// Looks up a signal number to get a basic string representation.
+///
+/// For known signals, this will typically give the same output as the C `strisignal`
+/// function does in the `C` locale.
+///
+/// Will return `None` for unknown or invalid signal numbers.
+///
+/// # Examples
+///
+/// ```
+/// #![feature(unix_signal_strings)]
+/// use std::os::unix::process::signal_string_raw;
+/// assert_eq!(signal_string_raw(15), Some("Terminated"));
```

Comment:
This relies on the assumption of which signal 15 is, *and* that the description supplied by the C library includes "Terminated". Is this a safe assumption to make in a doctest that may run on an arbitrary non-Linux UNIX target? Do we want a test failure if this assumption doesn't hold?

I think the number is portable to at least every Linux target. If the description is portable to every UNIX target we currently have, this may be OK, though we may not want to encourage people to rely on the exact output anyway.

Reply:
> This relies on the assumption of which signal 15 is, _and_ that the description supplied by the C library includes "Terminated". Is this a safe assumption to make in a doctest that may run on an arbitrary non-Linux UNIX target?

Yes, I think it is a pretty safe assumption to make on an arbitrary non-Linux Unix target.

Historically, when one wanted to send a particular signal with `kill(1)` it was necessary to pass the signal number in, not an abbreviation (so `kill -15 <pid>` rather than `kill -TERM <pid>`).  Many people's finger macros do this habitually because it's shorter :-).  I think at least we can rely on `9` and `15` being `Killed` and `Terminated`.

The assumption about the string is more questionable but I think use of some other string than `Terminated` will be extremely rare.

> Do we want a test failure if this assumption doesn't hold?

Well, obviously it woudl be best to avoid that, but it is difficult to check that this function is working *at all* without making *some* assumption about the output.  In practice I expect this assumption to be true everywhere.  (I'm an old Unix hand and have extensive experience of ancient proprietary unices.)

If it is not true somewhere then we might need to make this test have multiple possible permitted outputs.  I think that is better than some more vague test case.

Reply:
If it's likely to pass everywhere, that seems fine.